src/marketplace/AccessoriesView.js

```diff
@@ -1,11 +1,117 @@
 //@flow strict
 import * as React from 'react';
-class AccessoriesView extends React.Component<{||}> {
-    render() {    
+import {getTankComponent, verifyComponent} from '../armory/GetInventoryInfo.js';
+import Cookies from 'universal-cookie';
+
+type Props = {||}; 
+type State = {|
+	userId: string,
+|};
+let itemsForSale = [];
+class AccessoriesView extends React.Component<Props, State> {
+    constructor() {
+		super();
+		this.state={
+			//This tank id is the one the user is currently working on
+			userId: '',
+		}
+		this.getMarketSales();
+		this.getUserId();
+	}
+
+	//This gets us the user's id 
+	getUserId = async ():Promise<void> => {
+		const cookies = new Cookies();
+		const token = cookies.get('token');
+		const response = await fetch('/api/user/getUser/', {
+			method: 'GET',
+			headers: {
+				'Access-Control-Allow-Origin': '*',
+				'Content-Type': 'application/json',
+				'Access-Control-Allow-Credentials': 'true',
+				'x-auth-token': token
+			},
+		});
+		const body = await response.text();
+		const jsonObjectOfUser = JSON.parse(body);
+		this.setState({userId:jsonObjectOfUser._id});
+    };
+
+	//Gets all the sells that we are looking at 
+    getMarketSales = async ():Promise<void> => {
+		itemsForSale = [];
+		const response = await fetch('/api/marketplace/getMarketSales/', {
+			method: 'GET',
+			headers: {
+				'Access-Control-Allow-Origin': '*',
+				'Content-Type': 'application/json',
+				'Access-Control-Allow-Credentials': 'true',
+			},
+		});
+		const body = await response.text();
+		const jsonObjectOfSells = JSON.parse(body);
+		for (const sale in jsonObjectOfSells)
+		{
+			const typeOfItem = getTankComponent(verifyComponent(jsonObjectOfSells[sale].itemId));
+			if(typeOfItem === 'item')
+			{
+				let obj = {};
+				obj['name'] = jsonObjectOfSells[sale].itemId;
+				obj['price'] = jsonObjectOfSells[sale].salePrice;
+				obj['amount'] = jsonObjectOfSells[sale].amount;
+				obj['sellerId'] = jsonObjectOfSells[sale].sellerId;
+				obj['saleId'] = jsonObjectOfSells[sale]._id;
+				itemsForSale.push(obj);
+			}
+		}   
+		//Need this to deal with the asynch nature of api calling......fun times
+		//Makes sure that we create the cards after we get the sells
+		this.forceUpdate();     
```

Comment:
You shouldn't need this if you are doing everything else right. The reason it isn't updating is because you are storing things in external variables (i. e. variables outside the component) which you shouldn't be, so react doesn't get notified when its state changes.

Reply:
In other words, you should delete this because if something is going wrong, it is much better to know about it than for us to think it works when it doesn't